Minor tweaks to primary.py: docstring, arg validation, removing unused code

awwad opened the pull request:
This is a small collection of a few minor adjustments to the `primary.py` module in the Reference Implementation:

- The addition/improvement of a docstring for `primary.register_ecu_manifest`
- The addition of a minor argument check that was missing from `primary.__init__()` (and the addition of a corresponding test)
- The removal of some commented-out code in `primary.py`

You may notice that the docstring indicates that `uptane.UnknownVehicle` is raised if the Primary doesn't know the VIN provided in the submission of the ECU Manifest, even though code on this branch still incorrectly raises `uptane.Error` in that situation. This is corrected in another PR, #135. They will be merged together. This seemed better than going to the effort of backtracking and adding commits on both sides.

awwad committed:
Remove commented-out code in primary.py

awwad committed:
DOC: Improve docstring for primary.register_ecu_manifest

awwad committed:
Add a missing argument format check for repo name to primary.__init__()

And update the relevant test code.

vladimir-v-diaz commented on uptane/clients/primary.py:
```diff
@@ -934,18 +927,69 @@ def _check_ecu_serial(self, ecu_serial):
+
+    <Exceptions>
+
+      uptane.Spoofing
```
I think this function can potentially raise a FormatError.  Also, why don't you validate all of the arguments (e.g., `vin`)?

awwad replied:
Thanks, good catch. Fixing

vladimir-v-diaz commented on uptane/clients/primary.py:
```diff
@@ -934,18 +927,69 @@ def _check_ecu_serial(self, ecu_serial):
+
     """
     # check arg format and that serial is registered
     self._check_ecu_serial(ecu_serial)
```
This can raise `FormatError`.

awwad committed:
Perform additional argument validation in primary.register_ecu_manifest

for vin, nonce, and signed_ecu_manifest.

awwad committed:
DOC: Improve docstring for primary.register_ecu_manifest:

include tuf.FormatError as possible exception and clarify/correct
wording for force_pydict and signed_ecu_manifest arguments.
